ports/servoshell/desktop/dialog.rs

```diff
@@ -302,6 +318,54 @@ impl Dialog {
                 });
                 is_open
             },
+            Dialog::SelectDevice {
+                devices,
+                selected_device_index,
+                response_sender,
+            } => {
+                let mut is_open = true;
+                let modal = Modal::new("device_picker".into());
+                modal.show(ctx, |ui| {
+                    let mut frame = egui::Frame::default().inner_margin(10.0).begin(ui);
+                    frame.content_ui.set_min_width(150.0);
+
+                    frame.content_ui.heading("Choose a Device");
+                    frame.content_ui.add_space(10.0);
+
+                    egui::ComboBox::from_label("")
+                        .selected_text(devices[*selected_device_index + 1].clone())
```

Comment:
I think we can actually avoid the `clone()` on this particular line by doing `.selected_text(&devices[*selected_device_index + 1])` instead.